lib/pbench/test/unit/server/test_datasets_compare.py

```diff
@@ -0,0 +1,196 @@
+
+
+def mock_get_value(dataset, key) -> str:
+    if dataset.name == "uperf_3" or dataset.name == "uperf_4":
+        return "hammerDB"
+    return "uperf"
+
+
+class TestCompareDatasets:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, get_token_func):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            get_token_func: Pbench token fixture
+        """
+
+        def query_api(
+            datasets: list, user: str, expected_status: HTTPStatus
+        ) -> requests.Response:
+            ds_list = []
+            for dataset in datasets:
+                try:
+                    dataset_id = Dataset.query(name=dataset).resource_id
+                    ds_list.append(dataset_id)
+                except DatasetNotFound:
+                    ds_list.append(dataset)  # Allow passing deliberately bad value
+            headers = None
+            if user:
+                headers = {"authorization": f"bearer {get_token_func(user)}"}
+            response = client.get(
+                f"{server_config.rest_uri}/compare",
+                query_string={"datasets": ds_list},
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    class MockTarball:
+        tarball_path = Path("/dataset/tarball.tar.xz")
+        name = "tarball"
+
+        def extract(_tarball_path: Path, _path: str) -> str:
+            return "CSV_file_as_a_string"
+
+    def mock_find_dataset(self, dataset) -> MockTarball:
+        # Validate the resource_id
+        Dataset.query(resource_id=dataset)
+        return self.MockTarball
+
+    def test_get_no_dataset(self, query_get_as, monkeypatch):
+        monkeypatch.setattr(Metadata, "getvalue", mock_get_value)
+
+        response = query_get_as(
+            ["uperf_1", "nonexistent-dataset"], "drb", HTTPStatus.BAD_REQUEST
+        )
+        assert (
+            response.json["message"]
+            == "Unrecognized list value ['nonexistent-dataset'] given for parameter datasets; expected Dataset"
+        )
+
+    def test_dataset_not_present(self, query_get_as, monkeypatch):
+        monkeypatch.setattr(Metadata, "getvalue", mock_get_value)
+
+        query_get_as(["fio_2"], "drb", HTTPStatus.INTERNAL_SERVER_ERROR)
+
+    def test_unauthorized_access(self, query_get_as, monkeypatch):
+        monkeypatch.setattr(Metadata, "getvalue", mock_get_value)
+
+        response = query_get_as(["uperf_1", "uperf_2"], "drb", HTTPStatus.FORBIDDEN)
+        assert response.json == {
+            "message": "User drb is not authorized to READ a resource owned by test with private access"
+        }
```

Comment:
It would be good to match only the message key here, instead of the whole dictionary (like you did at line 121 and elsewhere).